Update for new modloader and HarmonyX. (Not tested, just fixed all errors)

lovhb opened the pull request:
Updated harmony and added the new modloader framework. Fixed all errors. Haven't tested though as I don't have a haptic vest.

lovhb committed:
Update for new modloader

Updated harmony and added the new framework.

MarshMello0 commented:
This fixes #1

MarshMello0 commented on .idea/.idea.vtol_bhaptics/.idea/discord.xml:
You might want to add `.idea/` to the `.gitignore` as people might have different prefences to these settings

MarshMello0 commented on vtol_bhaptics/vtol_bhaptics.csproj:
```diff
@@ -31,30 +31,38 @@
+      <HintPath>..\Bhaptics_integration\Dependencies\Bhaptics.Tact.dll</HintPath>
     </Reference>
-    <Reference Include="ModLoader">
-      <HintPath>..\Dependencies\ModLoader.dll</HintPath>
+    <Reference Include="Mod Loader">
+      <HintPath>C:\Program Files (x86)\Steam\steamapps\common\VTOL VR\@Mod Loader\Managed\Mod Loader.dll</HintPath>
+    </Reference>
+    <Reference Include="ModLoader.Framework">
+      <HintPath>C:\Program Files (x86)\Steam\steamapps\common\VTOL VR\@Mod Loader\Managed\ModLoader.Framework.dll</HintPath>
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Core" />
     <Reference Include="UnityEngine">
-      <HintPath>..\Dependencies\UnityEngine.dll</HintPath>
+      <HintPath>C:\Program Files (x86)\Steam\steamapps\common\VTOL VR\VTOLVR_Data\Managed\UnityEngine.dll</HintPath>
     </Reference>
     <Reference Include="UnityEngine.CoreModule">
-      <HintPath>..\Dependencies\UnityEngine.CoreModule.dll</HintPath>
+      <HintPath>C:\Program Files (x86)\Steam\steamapps\common\VTOL VR\VTOLVR_Data\Managed\UnityEngine.CoreModule.dll</HintPath>
     </Reference>
     <Reference Include="UnityEngine.PhysicsModule">
-      <HintPath>..\Dependencies\UnityEngine.PhysicsModule.dll</HintPath>
```
You also might want to change these back as not everyone has the game installed on their C drive. But everyone can make mklinks from the Dependencies folder to their game folder and not have to commit over each other. 😄

lovhb replied:
Oh i didn't realize rider uses absolute file paths for references... will fix

lovhb committed:
Update .gitignore

lovhb committed:
Updated .csproj to relative mklink directories

Dependencies/VTOL is for VTOL VR .dll's
Dependencies/ModLoader is for Modloader .dll's

lovhb committed:
Update .gitignore

lovhb committed:
Deleted some .idea files

lovhb committed:
Update .gitignore

lovhb committed:
Updated .tact files location and .NET 4.8

Updated to point to mod folder instead of old modloader directory.
Changed .NET version to 4.8 to work with VTOLAPI

lovhb commented:
Update: The mod now works in game. The only remaining issue is that gun_fire rumble is apparently slightly delayed.

McFredward commented:
Just tested it - it's working. Thanks!
This community is so nice - the problem just got fixed before I even had time to look at it haha.

Will try to fix some old bugs and bugs which appeared because of the games updates the next days.
